lib/iris/util.py

```diff
@@ -936,16 +936,26 @@ def format_array(arr):
     For customisations, use the :mod:`numpy.core.arrayprint` directly.
 
     """
-    if arr.size > 85:
-        summary_insert = "..., "
-    else:
-        summary_insert = ""
+
+    summary_insert = ""
+    summary_threshold = 85
+    edge_items = 3
     ffunc = str
-    return np.core.arrayprint._formatArray(arr, ffunc, len(arr.shape),
-                                           max_line_len=50,
-                                           next_line_prefix='\t\t',
-                                           separator=', ', edge_items=3,
-                                           summary_insert=summary_insert)[:-1]
+    formatArray = np.core.arrayprint._formatArray
+    max_line_len = 50
+    legacy = '1.13'
+    if arr.size > summary_threshold:
+        summary_insert = '...'
+    options = np.get_printoptions()
+    options['legacy'] = legacy
```

Comment:
Do we really need to do this, if virtually every CML is changing anyway ??

N.B. I have no idea how much more carnage that might be !!
So far (https://github.com/bjlittle/iris/tree/d910238b43bf3e5b57ff35c4ac813d7a077626c9/  ~= 9am Thu 17 May) we have 77 CML files changed out of 569 ???

Reply:
Yes, we really need to do this.

Reply:
To put this in context... I'm down to my last 17 test failures and 1 error (excluding graphical tests), so I'm in the home straight. I don't expect another ~500 CML file changes :pray:

Reply:
The graphical tests will be a different can of worms...

Reply:
I agree with merging this as it is. I'd like to follow it up with another PR though and change all ~500 CML to modern numpy.